java/flight/flight-core/src/main/java/org/apache/arrow/flight/client/ClientCookieMiddleware.java

```diff
@@ -37,29 +37,36 @@
  */
 public class ClientCookieMiddleware implements FlightClientMiddleware {
   private static final String SET_COOKIE_HEADER = "Set-Cookie";
-  private static final String SET_COOKIE2_HEADER = "Set-Cookie2";
   private static final String COOKIE_HEADER = "Cookie";
+  private final Factory factory;
 
-  // Use a map to track the most recent version of a cookie from the server.
-  // Note that cookie names are case-sensitive (but header names aren't).
-  private Map<String, HttpCookie> cookies = new HashMap<>();
-
-  public ClientCookieMiddleware() {
+  public ClientCookieMiddleware(Factory factory) {
```

Comment:
Let's make this package private instead of public.

Reply:
And mark @VisibleForTesting

Reply:
Done.